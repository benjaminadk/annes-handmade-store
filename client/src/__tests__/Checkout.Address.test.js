import React from 'react'
import Address from '../components/Checkout.Address'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

const ships = [
  {
    city: 'San Pedro',
    email: 'benjaminadk@gmail.com',
    firstName: 'Anne',
    id: '5b30dd5d5745dc0f581af7dc',
    lastName: 'Brooke',
    notes:
      'Make sure all the buttons are in the proper alignment. Make sure all of the site is kosher on mobile screens.',
    state: 'California',
    street1: '3240 S Kerckhoff Ave',
    street2: '',
    title: 'Home',
    zip: '90731'
  }
]

describe('Checkout Address Component', () => {
  it('renders correctly with empty ships prop', () => {
    const tree = renderer.create(<Address ships={[]} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders correctly with a ships prop', () => {
    const wrapper = shallow(<Address ships={ships} />)
    expect(wrapper.props().ships).toEqual(ships)
  })
})
