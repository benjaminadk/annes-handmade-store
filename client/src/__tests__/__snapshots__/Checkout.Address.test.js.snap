// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkout Address Component renders correctly with empty ships prop 1`] = `
<div
  className="Address-noAddress-8"
>
  <h1
    className="MuiTypography-root-10 MuiTypography-display2-13 MuiTypography-alignCenter-23"
  >
    Please Provide A Shipping Address
  </h1>
  <button
    className="MuiButtonBase-root-56 MuiButton-root-34 MuiButton-contained-43 MuiButton-containedSecondary-45 MuiButton-raised-46 MuiButton-raisedSecondary-48 MuiButton-sizeLarge-54 Address-noAddressButton-9"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-35"
    >
      Add New Address
    </span>
    <span
      className="MuiTouchRipple-root-59"
    />
  </button>
</div>
`;
